third_party/amd/lib/TritonAMDGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -431,6 +431,16 @@ static Value convertBf16ToFp32(Location loc,
   return b.bitcast(shifted, f32_ty);
 }
 
+static Value checkNan(Location loc, RewriterBase &rewriter, Value v) {
```

Comment:
Can you name this as `checkIsNaN` and also use it for fp16 to fp8 conversion? We can pass in a `TritonLLVMOpBuilder &` to reuse instead of recreating it (which can break insertion points).

Reply:
Fixed.

Reply:
For "use it for fp16 to fp8 conversion", we have hardware instruction for the conversion on MI300, so where should we use it?